### PR TITLE
feat(atom/tooltip): fix children element when include component

### LINKED ISSUE #1223: Combine AtomButton with AtomTooltip does not work
**Describe the bug**

When wrapping an AtomButton inside an AtomTooltip directly, like this

```   
    <AtomTooltip>
      <AtomButton onClick={onClick} size="small" design="flat" title="HERE the TOOLTIP">
        <AtomIcon color={color}>{icon}</AtomIcon>
      </AtomButton>
    </AtomTooltip>
```
The components cannot render and will display several errors

**Steps to reproduce**
1. Initialize the Sui-Studio with the code
2. It will display the error: _TypeError: Cannot read property 'removeEventListener' of null_

**Expected behaviour**
The Components tooltip and button should be render properly.

**Environment / Context**
-Browser Chrome 85

**Additional info**
According to the documentation: https://sui-components.now.sh/workbench/atom/tooltip/documentation/readme

it will show the title attribute of the wrapped element by default
tooltip can be applied over buttons (or any element w/ onClick)

I can do apply it over a HTML <button> but not with the <AtomButton>

Full error report: 

```
TypeError: Cannot read property 'removeEventListener' of null
AtomTooltip.componentWillUnmount
http://localhost:3000/main.js:2521:12
callComponentWillUnmountWithTimer
http://localhost:3000/main.js:59808:12
HTMLUnknownElement.callCallback
http://localhost:3000/main.js:40416:14
invokeGuardedCallbackDev
http://localhost:3000/main.js:40465:16
invokeGuardedCallback
http://localhost:3000/main.js:40520:31
safelyCallComponentWillUnmount
http://localhost:3000/main.js:59815:5
commitUnmount
http://localhost:3000/main.js:60337:11
commitNestedUnmounts
http://localhost:3000/main.js:60391:5
unmountHostComponents
http://localhost:3000/main.js:60671:7
commitDeletion
http://localhost:3000/main.js:60728:5
commitMutationEffects
http://localhost:3000/main.js:63010:11
HTMLUnknownElement.callCallback
http://localhost:3000/main.js:40416:14
invokeGuardedCallbackDev
http://localhost:3000/main.js:40465:16
invokeGuardedCallback
http://localhost:3000/main.js:40520:31
commitRootImpl
http://localhost:3000/main.js:62737:9
unstable_runWithPriority
http://localhost:3000/main.js:69250:12
runWithPriority$1
http://localhost:3000/main.js:51267:10
commitRoot
http://localhost:3000/main.js:62609:3
finishSyncRender
http://localhost:3000/main.js:62035:3
performSyncWorkOnRoot
http://localhost:3000/main.js:62021:7
(anonymous function)
http://localhost:3000/main.js:51317:24
unstable_runWithPriority
http://localhost:3000/main.js:69250:12
runWithPriority$1
http://localhost:3000/main.js:51267:10
flushSyncCallbackQueueImpl
http://localhost:3000/main.js:51312:7
flushSyncCallbackQueue
http://localhost:3000/main.js:51300:3
batchedUpdates$1
http://localhost:3000/main.js:62090:7
act
http://localhost:3000/main.js:39664:14
render
http://localhost:3000/main.js:23155:3
setup
http://localhost:3000/3.js:34:81
Context.<anonymous>
http://localhost:3000/3.js:43:18
callFn
lib/runnable.js:358:20
  355 | }
  356 | 
  357 | function callFn(fn) {
> 358 |   var result = fn.call(ctx);
      |                  ^  359 |   if (result && typeof result.then === 'function') {
  360 |     self.resetTimeout();
  361 |     result.then(
View compiled
Test.Runnable.run
lib/runnable.js:346:4
  343 | 
  344 | // sync or promise-returning
  345 | try {
> 346 |   callFn(this.fn);
      |  ^  347 | } catch (err) {
  348 |   errorWasHandled = true;
  349 |   if (err instanceof Pending) {
View compiled
Runner.runTest
lib/runner.js:621:9
  618 |   return test.run(fn);
  619 | }
  620 | try {
> 621 |   test.run(fn);
      |       ^  622 | } catch (err) {
  623 |   fn(err);
  624 | }
View compiled
(anonymous function)
lib/runner.js:745:11
  742 |   return hookErr(err, errSuite, false);
  743 | }
  744 | self.currentRunnable = self.test;
> 745 | self.runTest(function(err) {
      |     ^  746 |   test = self.test;
  747 |   // conditional skip within it
  748 |   if (test.pending) {
View compiled
next
lib/runner.js:538:13
  535 | 
  536 | if (!suite) {
  537 |   self.suite = orig;
> 538 |   return fn();
      |         ^  539 | }
  540 | 
  541 | self.hook(name, function(err) {
View compiled
(anonymous function)
lib/runner.js:548:6
  545 |       return fn(err, errSuite);
  546 |     }
  547 | 
> 548 |     next(suites.pop());
      |    ^  549 |   });
  550 | }
  551 | 
View compiled
next
lib/runner.js:430:13
  427 | function next(i) {
  428 |   var hook = hooks[i];
  429 |   if (!hook) {
> 430 |     return fn();
      |           ^  431 |   }
  432 |   self.currentRunnable = hook;
  433 | 
View compiled
(anonymous function)
lib/runner.js:516:4
  513 |   }
  514 | 
  515 |   Runner.immediately(function() {
> 516 |     next(0);
      |    ^  517 |   });
  518 | };
  519 | 
View compiled
timeslice
browser-entry.js:100:4
   97 | function timeslice() {
   98 |   var immediateStart = new Date().getTime();
   99 |   while (immediateQueue.length && new Date().getTime() - immediateStart < 100) {
> 100 |     immediateQueue.shift()();
      |    ^  101 |   }
  102 |   if (immediateQueue.length) {
  103 |     immediateTimeout = setTimeout(timeslice, 0);
View compiled
```

```
TypeError: Cannot read property 'addEventListener' of null
(anonymous function)
http://localhost:3000/main.js:2491:21
AtomTooltip.componentDidMount
http://localhost:3000/main.js:2490:33
commitLifeCycles
http://localhost:3000/main.js:60042:22
commitLayoutEffects
http://localhost:3000/main.js:63031:7
HTMLUnknownElement.callCallback
http://localhost:3000/main.js:40416:14
invokeGuardedCallbackDev
http://localhost:3000/main.js:40465:16
invokeGuardedCallback
http://localhost:3000/main.js:40520:31
commitRootImpl
http://localhost:3000/main.js:62769:9
unstable_runWithPriority
http://localhost:3000/main.js:69250:12
runWithPriority$1
http://localhost:3000/main.js:51267:10
commitRoot
http://localhost:3000/main.js:62609:3
finishSyncRender
http://localhost:3000/main.js:62035:3
performSyncWorkOnRoot
http://localhost:3000/main.js:62021:7
scheduleUpdateOnFiber
http://localhost:3000/main.js:61416:7
updateContainer
http://localhost:3000/main.js:64601:3
(anonymous function)
http://localhost:3000/main.js:64986:7
unbatchedUpdates
http://localhost:3000/main.js:62131:12
legacyRenderSubtreeIntoContainer
http://localhost:3000/main.js:64985:5
render
http://localhost:3000/main.js:65068:10
(anonymous function)
http://localhost:3000/main.js:23159:56
batchedUpdates$1
http://localhost:3000/main.js:62084:12
act
http://localhost:3000/main.js:39664:14
render
http://localhost:3000/main.js:23155:3
setup
http://localhost:3000/3.js:34:81
Context.<anonymous>
http://localhost:3000/3.js:43:18
callFn
lib/runnable.js:358:20
  355 | }
  356 | 
  357 | function callFn(fn) {
> 358 |   var result = fn.call(ctx);
      |                  ^  359 |   if (result && typeof result.then === 'function') {
  360 |     self.resetTimeout();
  361 |     result.then(
View compiled
Test.Runnable.run
lib/runnable.js:346:4
  343 | 
  344 | // sync or promise-returning
  345 | try {
> 346 |   callFn(this.fn);
      |  ^  347 | } catch (err) {
  348 |   errorWasHandled = true;
  349 |   if (err instanceof Pending) {
View compiled
Runner.runTest
lib/runner.js:621:9
  618 |   return test.run(fn);
  619 | }
  620 | try {
> 621 |   test.run(fn);
      |       ^  622 | } catch (err) {
  623 |   fn(err);
  624 | }
View compiled
(anonymous function)
lib/runner.js:745:11
  742 |   return hookErr(err, errSuite, false);
  743 | }
  744 | self.currentRunnable = self.test;
> 745 | self.runTest(function(err) {
      |     ^  746 |   test = self.test;
  747 |   // conditional skip within it
  748 |   if (test.pending) {
View compiled
next
lib/runner.js:538:13
  535 | 
  536 | if (!suite) {
  537 |   self.suite = orig;
> 538 |   return fn();
      |         ^  539 | }
  540 | 
  541 | self.hook(name, function(err) {
View compiled
(anonymous function)
lib/runner.js:548:6
  545 |       return fn(err, errSuite);
  546 |     }
  547 | 
> 548 |     next(suites.pop());
      |    ^  549 |   });
  550 | }
  551 | 
View compiled
next
lib/runner.js:430:13
  427 | function next(i) {
  428 |   var hook = hooks[i];
  429 |   if (!hook) {
> 430 |     return fn();
      |           ^  431 |   }
  432 |   self.currentRunnable = hook;
  433 | 
View compiled
(anonymous function)
lib/runner.js:516:4
  513 |   }
  514 | 
  515 |   Runner.immediately(function() {
> 516 |     next(0);
      |    ^  517 |   });
  518 | };
  519 | 
View compiled
timeslice
browser-entry.js:100:4
   97 | function timeslice() {
   98 |   var immediateStart = new Date().getTime();
   99 |   while (immediateQueue.length && new Date().getTime() - immediateStart < 100) {
> 100 |     immediateQueue.shift()();
      |    ^  101 |   }
  102 |   if (immediateQueue.length) {
  103 |     immediateTimeout = setTimeout(timeslice, 0);
View compiled
```

```
Unhandled Rejection (TypeError): Failed to execute 'observe' on 'IntersectionObserver': parameter 1 is not of type 'Element'.
▶ 32 stack frames were collapsed.
callFn
lib/runnable.js:358:20
  355 | }
  356 | 
  357 | function callFn(fn) {
> 358 |   var result = fn.call(ctx);
      |                  ^  359 |   if (result && typeof result.then === 'function') {
  360 |     self.resetTimeout();
  361 |     result.then(
View compiled
Test.Runnable.run
lib/runnable.js:346:4
  343 | 
  344 | // sync or promise-returning
  345 | try {
> 346 |   callFn(this.fn);
      |  ^  347 | } catch (err) {
  348 |   errorWasHandled = true;
  349 |   if (err instanceof Pending) {
View compiled
Runner.runTest
lib/runner.js:621:9
  618 |   return test.run(fn);
  619 | }
  620 | try {
> 621 |   test.run(fn);
      |       ^  622 | } catch (err) {
  623 |   fn(err);
  624 | }
View compiled
(anonymous function)
lib/runner.js:745:11
  742 |   return hookErr(err, errSuite, false);
  743 | }
  744 | self.currentRunnable = self.test;
> 745 | self.runTest(function(err) {
      |     ^  746 |   test = self.test;
  747 |   // conditional skip within it
  748 |   if (test.pending) {
View compiled
next
lib/runner.js:538:13
  535 | 
  536 | if (!suite) {
  537 |   self.suite = orig;
> 538 |   return fn();
      |         ^  539 | }
  540 | 
  541 | self.hook(name, function(err) {
View compiled
(anonymous function)
lib/runner.js:548:6
  545 |       return fn(err, errSuite);
  546 |     }
  547 | 
> 548 |     next(suites.pop());
      |    ^  549 |   });
  550 | }
  551 | 
View compiled
next
lib/runner.js:430:13
  427 | function next(i) {
  428 |   var hook = hooks[i];
  429 |   if (!hook) {
> 430 |     return fn();
      |           ^  431 |   }
  432 |   self.currentRunnable = hook;
  433 | 
View compiled
(anonymous function)
lib/runner.js:516:4
  513 |   }
  514 | 
  515 |   Runner.immediately(function() {
> 516 |     next(0);
      |    ^  517 |   });
  518 | };
  519 | 
View compiled
timeslice
browser-entry.js:100:4
   97 | function timeslice() {
   98 |   var immediateStart = new Date().getTime();
   99 |   while (immediateQueue.length && new Date().getTime() - immediateStart < 100) {
> 100 |     immediateQueue.shift()();
      |    ^  101 |   }
  102 |   if (immediateQueue.length) {
  103 |     immediateTimeout = setTimeout(timeslice, 0);
```

### PR DESCRIPTION
**Pull request by klaufel**
ISSUES CLOSED: #1223

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Description, Motivation and Context
Resolved issue https://github.com/SUI-Components/sui-components/issues/1223

Fixed issue with adding components without having to have a top container. In this way, if the Tooltip receives an element / component node has wrapper will be automatically.

Inside the children prop that it receives, a modifier with `display: inline-flex` is added to make it fit the content with which it is included.

```
<AtomTooltip>
  <AtomButton
    title="tooltip content"
    onClick={() => console.log('Hi!')}
  >
    Component
  </AtomButton>
</AtomTooltip>
```

### Screenshots - Animations

CleanShot 2020-10-01 at 17 49 48

### CODE REVIEW

**Review by xaviermariaconejo — APPROVED** [review]

File: components/atom/tooltip/src/index.js [hunk 1]
```diff
@@ -61,11 +61,18 @@ class AtomTooltip extends Component {
     return React.Children.map(childrenOnly, child => {
       this.onClickTarget = child.props.onClick
       this.title = child.props.title
-      return React.cloneElement(child, {
-        ref,
-        className,
-        onTouchEnd
-      })
+
+      return typeof child?.type !== 'string'
```
Comment (midudev) [inline components/atom/tooltip/src/index.js]: Bastante raro hacer un `child?` aquí para asegurarte que existe `child` cuando en las dos líneas anteriores se accede a child sin control para acceder a sus props.

**Review by adriadoz — APPROVED** [review]

File: components/atom/tooltip/src/index.js [hunk 2]
```diff
@@ -61,11 +61,18 @@ class AtomTooltip extends Component {
...
+        ? React.createElement(
+            'div',
+            {
+              ref,
+              className: cx(className, `${className}--wrapper`),
```
Comment (nucliweb) [inline components/atom/tooltip/src/index.js]: I think that you don't need `classnames` to concatenate two classes.

```suggestion
              className: `${className} ${className}--wrapper`,
```

Comment (klaufel) [reply components/atom/tooltip/src/index.js]: yes, is not necessary used with CX, thanks 🤗

File: components/atom/tooltip/src/index.js (hunk 1 above)
Comment (klaufel) [reply components/atom/tooltip/src/index.js]: Actualizado, gracias!

**Review by andresin87 — APPROVED** [review]

**Review by midudev — APPROVED** [review]

**Review by desko27 — APPROVED** [review]

**Pull request merged**